bin/yaml/cpp.yaml

```diff
@@ -758,6 +758,7 @@ compilers:
         - 14.0.0
         - 14.0.5
         - 16.0.0
+        - trunk
```

Comment:
in this way it'll only get installed once: the `trunk` stuff needs to be done slightly differently in order to not fall afoul of our "don't reinstall things". Plus trunk builds are actually dated... I'll fix up in a separate change; but thanks!